docs/example/pytest.rst

```diff
@@ -24,7 +24,7 @@ and the following ``tox.ini`` content:
 
     [testenv]
     deps = pytest               # PYPI package providing pytest
-    commands = pytest {posargs} # substitute with tox' positional arguments
+    commands = python -m pytest {posargs} # substitute with tox' positional arguments
```

Comment:
if this doesn't work with (unsupported) tox-current-env that feels to me a bug on their part.

Reply:
Yes, this could be saw as a bug on their side, or simply design choice (the latter seems to be true).  BTW, many projects already does this, for example [mypy](https://github.com/python/mypy/blob/master/tox.ini#L19), [wheel](https://github.com/pypa/wheel/blob/main/tox.ini#L13), or [Pillow](https://github.com/python-pillow/Pillow/blob/main/tox.ini#L19).

Reply:
their project is unsupported and there's no improvement to doing it this way in tox. if current-env doesn't adjust PATH properly I don't think we should make our docs worse to accommodate them

Reply:
btw those are done that way to pass -W, not for any other reason (and that can slash should be done with PYTHONWARNINGS or the pytest setting)

Reply:
Could you please elaborate how this would make the documentation worse?

The problem is not in `PATH`.  Let me show you real example where I need to deal with this.

In [OpenIndiana](https://www.openindiana.org/) we support several Python versions concurrently with one version considered as default.  Currently we support Python 3.7 and 3.9.  This means that we deliver `/usr/bin/python3.7` and `/usr/bin/python3.9` together with `/usr/bin/tox-3.7`, `/usr/bin/tox-3.9` and `/usr/bin/pytest-3.7`, `/usr/bin/pytest-3.9`.  There is also a symlink pointing to the default version, so we do have `/usr/bin/python -> python3.9`, `/usr/bin/tox -> tox-3.9`, `/usr/bin/pytest -> pytest-3.9`.

Now, when we want to test something for non-default Python version (e.g. 3.7), we just run `tox-3.7 --current-env`, then the `tox-current-env` plugin prepares the testing environment by creating symlink `python -> /usr/bin/python3.7`.  I think nothing else is prepared (by design).  When `tox` tries to run `pytest` binary directly it simply cannot find version 3.7 of `pytest` because the only `pytest` in the system is symlink to `pytest-3.9` and there is no way to solve that via `PATH`.  The only reasonable way to find `pytest-3.7` is via `python -m pytest`.

In addition, the `python -m pytest` way of invocation `pytest` is officially documented and supported way of running `pytest`, please see https://docs.pytest.org/en/latest/how-to/usage.html#calling-pytest-through-python-m-pytest.

Thank you.

Reply:
I'm a pytest core dev. we should not cargo cult this unnecessarily for an unsupported plugin

Reply:
> btw those are done that way to pass -W, not for any other reason (and that can slash should be done with PYTHONWARNINGS or the pytest setting)

Sorry, but mypy does not pass -W.  Also, to pass -W to pytest you do not need to invoke it via `python -m pytest -W ...`, you can do direct `pytest -W ...` (at least `pytest --help` suggests so).

Reply:
yes that's why I said unnecessarily, but if you take a look at the git history that's why it was changed in those projects